tests/test_handlers.py

```diff
@@ -45,4 +69,36 @@ async def test_basic_status_wrong_id(self):
         resp = await request.json()
         assert "no such job foobar found" in resp["msg"].lower()
 
+    async def test_basic_status_successful_job(self):
+        with mock.patch.object(archive_verify.pdc_client.PdcClient, "download") as download_mock, \
+                mock.patch("archive_verify.workers.compare_md5sum") as md5_mock:
+            download_mock.return_value = True
+            md5_mock.return_value = True
+
+            request = await self.post_queued_request(endpoint="download")
+
+            assert request.status == 200
+            resp = await request.json()
+
+            url = self.BASE_URL + "/status/" + resp["job_id"]
+            request = await self.client.request("GET", url)
+            assert request.status == 200
+            resp = await request.json()
+            assert resp["state"] == "done"
+
+    async def test_basic_status_failed_job(self):
+        with mock.patch.object(archive_verify.pdc_client.PdcClient, "download") as download_mock, \
+                mock.patch("archive_verify.workers.compare_md5sum") as md5_mock:
+            download_mock.return_value = False
+            md5_mock.return_value = True
```

Comment:
Wouldn't it make more sense that the md5sum check fails if the download fails?

Reply:
in reality, the code will return the error immediately if the download fails (and that is what this test asserts) and not do the md5 check, so it doesn't matter what the mock returns. I can change it if it is confusing

Reply:
Ah, do we have to mock the function at all then? But if yes, I think it would make more sense to set it to false.

Reply:
yeah, not sure if it's a leftover or part of a grander plan that I abandoned 🤔. I'll clean it up a bit.